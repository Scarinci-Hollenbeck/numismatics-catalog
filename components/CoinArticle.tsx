import { create } from 'domain';
import React from 'react';
import { Article } from '../interfaces';
import ButtonLink from './ButtonLink';
import { createMarkup, formatDate } from '../utils/helpers';


type Props = {
  article: Article
}

export default function CoinArticle({ article }: Props): JSX.Element {

  return (
    <li>      
      <div>
        <h5>{article.node.title}</h5>
        <p><strong>Published: </strong>{formatDate(article.node.date)}</p>
        <div
          className="excerpt"
          dangerouslySetInnerHTML={createMarkup(article.node.excerpt)}
        />
        <ButtonLink
          link={`https://donaldscarinci.com${article.node.uri}`}
          caption="Full Article"
        />
      </div>
      <style jsx>
        {`
          li {
            font-family: 'Tajawal Regular';
            list-style-type: none;
            margin: 0;
            margin-bottom: 2em;
            padding: 0;
            text-align: left;
            padding-bottom: 2em;
            color: #fff;            
          }  

          .excerpt {
            margin-left: 0;
            margin-bottom: 2em;
          }

          h5 {
            font-size: 1.7rem;
          }

          p,
          h5 {
            margin: 0;
            padding: 0;
          }

          a {
            font-weight: bold;
            font-size: 1.2rem;
          }

          @media (min-width: 1225px) {
            li {
              margin-right: 3em;
            }
          }
        `}
      </style>
    </li>
  );
}
