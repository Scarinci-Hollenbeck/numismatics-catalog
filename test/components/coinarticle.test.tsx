import React from 'react';
import renderer from 'react-test-renderer';
import CoinArticle from '../../components/CoinArticle';

const articles = [
  {
    node: {
      id: '123',
      title: 'Austria’s Named Best Gold Coin at 2020 COTY Awards',
      excerpt:
        'The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
      featuredImage: {
        node: {
          sourceUrl:
            'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
        },
      },
      uri:
        'https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/',
    },
  },
  {
    node: {
      id: '456',
      title: 'Germany’s Named Best Gold Coin at 2021 COTY Awards',
      excerpt:
        'The Germany Mint won the “Best Gold Coin” category at 2021 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
      featuredImage: {
        node: {
          sourceUrl:
            'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
        },
      },
      uri:
        'https://donaldscarinci.com/germanys-named-best-gold-coin-at-2021-coty-awards/',
    },
  },
];

describe('Coin Article Component', () => {
  it('expect articles length == 2', () => {
    expect(articles.length).toBe(2);
  });

  it('The word "Coin" is found in the title', () => {
    const { title } = articles[0].node;
    expect(title).toMatch(/Coin/);
  });

  it('the coin article list contains sourceUrl and title', () => {
    expect(articles[1].node).toHaveProperty(
      'title',
      'Germany’s Named Best Gold Coin at 2021 COTY Awards',
    );
    expect(articles[1].node.featuredImage.node).toHaveProperty(
      'sourceUrl',
      'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
    );
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<CoinArticle article={articles[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
