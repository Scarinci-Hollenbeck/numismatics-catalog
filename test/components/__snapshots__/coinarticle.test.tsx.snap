// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Article Component matches snapshot 1`] = `
<li
  className="jsx-291405452"
>
  <div
    className="jsx-291405452"
  >
    <h5
      className="jsx-291405452"
    >
      Austria’s Named Best Gold Coin at 2020 COTY Awards
    </h5>
    <p
      className="jsx-291405452"
    >
      <strong
        className="jsx-291405452"
      >
        Published: 
      </strong>
      undefined NaN, NaN
    </p>
    <div
      className="jsx-291405452 excerpt"
      dangerouslySetInnerHTML={
        Object {
          "__html": "The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in",
        }
      }
    />
    <a
      className="buttonlink"
      href="https://donaldscarinci.comhttps//donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards"
      onClick={[Function]}
      onMouseEnter={[Function]}
      type="button"
    >
      Full Article
    </a>
  </div>
</li>
`;
